app/models/form.rb

```diff
@@ -5,20 +5,38 @@ class Form
 
   include Form::Setup
 
-  def initialize(form_path, name)
-    raise "No form definition file exists for given year".freeze unless File.exist?(form_path)
-
-    @name = name
-    @setup_sections = [Form::Setup::Sections::Setup.new(nil, nil, self)]
-    @form_definition = JSON.parse(File.open(form_path).read)
-    @form_sections = form_definition["sections"].map { |id, s| Form::Section.new(id, s, self) }
-    @type = form_definition["form_type"]
-    @sections =  setup_sections + form_sections
-    @subsections = sections.flat_map(&:subsections)
-    @pages = subsections.flat_map(&:pages)
-    @questions = pages.flat_map(&:questions)
-    @start_date = Time.iso8601(form_definition["start_date"])
-    @end_date = Time.iso8601(form_definition["end_date"])
+  def initialize(form_path, name, sections_in_form = [], type = "lettings")
+    if type == "sales"
+      # @form_definition = sections.map do |s|
+      #   s.new(nil, nil, self)
+      # end
+      @name = name
+      # @setup_sections = [Form::Setup::Sections::Setup.new(nil, nil, self)]
+      # @form_definition = JSON.parse(File.open(form_path).read)
+      @form_sections = sections_in_form.map { |sec| sec.new(nil, nil, self) }
+      @type = "sales"
+      @sections = form_sections
+      @subsections = sections.flat_map(&:subsections)
+      @pages = subsections.flat_map(&:pages)
+      @questions = pages.flat_map(&:questions)
+      @start_date = Time.zone.local(name[0..3], 4, 1)
+      @end_date = Time.zone.local(start_date.year + 1, 4, 1)
```

Comment:
This doesn't actually seem like it's necessarily true since they get extra time for late submissions and it seems that date moves more ad-hoc than we thought.

Reply:
If we keep it in we should probably at least match the lettings form end date? (1st July)

Reply:
https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/858/commits/04075e99fea8183ae556143c8821774134cdee4c